### PR TITLE
Use reqwest

### PR DESCRIPTION
**Pull request by yajirobee**
current module doesn't compile with openssl 1.1 as relying on rust-openssl v0.7.
To use openssl 1.1, rust-openssl should be 0.9 at least. 
https://github.com/sfackler/rust-openssl/releases/tag/v0.9.0

It means `hyper`'s version should be bumped. It seems that `hyper` changed interfaces to provide async IO features and it is recommended to use `reqwest` if only simple sync IO client is required.
Therefore, I updated dependencies to use `reqwest` instead of `hyper`.

### CODE REVIEW

File: src/client.rs [hunk 1]
```diff
@@ -512,19 +485,27 @@ impl <R> Client <R> where R: RequestExecutor {
...
-        let content_len = content_len as usize;
-
-        Ok((response, content_len))
+        let content_length = match response.headers().get(CONTENT_LENGTH) {
+            Some(header_value) => match header_value.to_str() {
+                Ok(ct_len_str) => match ct_len_str.parse::<usize>() {
+                    Ok(ct_len) => Ok(ct_len),
+                    _ => Err(TreasureDataError::ApiError(
+                        response.status(),
+                        "Failed to parse Content-Length header".to_string()))
```
Comment (komamitsu) [inline src/client.rs]: How about showing the actual value in error message?

File: Cargo.toml [hunk 2]
```diff
@@ -11,11 +11,14 @@ keywords = ["TreasureData"]
 
 [dependencies]
 log = "0.3"
-hyper = "0.9"
+reqwest = "0.9"
```
Comment (komamitsu) [inline Cargo.toml:14]: This library introduces `encoding_rs` which doesn't seem to support old Rust version https://github.com/hsivonen/encoding_rs/blob/master/README.md#rust-version-compatibility. Actually I failed to compile with 1.27.0...

Also, I ran into `reqwest`'s compile issue with 1.29.0. And I finally built this project with 1.30.0.

Could you add the note in README?

Comment (yajirobee) [reply Cargo.toml:14]: good catch. I didn't check rust compatibility. created `Requirements` section in README.
jfyi: I tested on 1.32.0 & 1.33.0(nightly).

**Review by komamitsu — APPROVED** [review]
LGTM 👍 
Thanks!

**Comment by komamitsu** [comment]
Released this as v0.7.1

**Pull request merged**